src/plugins.ts

```diff
@@ -0,0 +1,82 @@
+import {PluginConfig, Pod} from './pod';
+import {join} from 'path';
+
+export default class Plugins {
+  static DefaultPluginsDir = 'plugins';
+  static DefaultPluginsFilename = 'plugin.js';
+
+  readonly pod: Pod;
+  readonly plugins: any[];
+
+  constructor(pod: Pod) {
+    this.pod = pod;
+    this.plugins = [];
+  }
+
+  /**
+   * Register plugins based on a plugin configuration. Normally from the amagaki.yaml file.
+   */
+  async registerPlugins(pluginConfigs?: Array<PluginConfig>) {
+    if (!pluginConfigs) {
+      return;
+    }
+
+    for (const pluginConfig of pluginConfigs) {
+      const localePluginPath = join(
+        Plugins.DefaultPluginsDir,
+        pluginConfig.key,
+        Plugins.DefaultPluginsFilename
+      );
+
+      // Check for local plugin definition.
+      if (this.pod.fileExists(localePluginPath)) {
+        const pluginImport = await import(
```

Comment:
There was some discussion in slack, but FYI `require()` is synchronous and may be better suited for loading any config files or plugins.

Reply:
Switched to use a sync `require`.